### PR TITLE
intel-one-mono: init at 1.0.0

### PR DESCRIPTION
**Pull request by dpc**
###### Description of changes

https://github.com/intel/intel-one-mono/tree/main

> Introducing Intel One Mono, an expressive monospaced font family that’s built with clarity, legibility, and the needs of developers in mind.
>
> It’s easier to read, and available for free, with an open-source font license.
>
> Intel One Mono also covers a wide range of over 200 languages using the Latin script. The Intel One Mono fonts are provided in four weights — Light, Regular, Medium, and Bold — with matching italics, and we are happy to share both an oﬃcial release of fonts ready to use as well as editable sources.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by felschr — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runHook postInstall
+  '';
+
+
+  meta = with lib; {
+    description = "Intel One Mono Typeface";
+    homepage = "https://github.com/intel/intel-one-mono";
+    changelog = "https://github.com/intel/intel-one-mono";
+    license = licenses.ofl;
+    maintainers = [ ];
```
[the remark this answers is not in the document]
Comment (dpc) [reply pkgs/data/fonts/intel-one-mono/default.nix:32]: Sure, I guess I can try. Do I need to do anything? I think I need to add myself somewhere?

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    url = "https://github.com/intel/intel-one-mono/releases/download/V${version}/ttf.zip";
+    sha256 = "sha256-4XE4HS25TlwXzAh41WJS3d7T2U/qO3Hm4zlvMcgEnpU=";
+    stripRoot = false;
+  };
+
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  doCheck = false;
+  dontFixup = true;
```
[the remark this answers is not in the document]
Comment (dpc) [reply pkgs/data/fonts/intel-one-mono/default.nix:18]: I copied it all from some other existing font. I have no idea.

**Review by drupol — APPROVED** [review]
LGTM !

File: pkgs/data/fonts/intel-one-mono/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/data/fonts/intel-one-mono/default.nix]: Yes, add yourself please.

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    install -Dm644 -t $out/share/fonts/truetype/ ttf/*.ttf
+    install -Dm644 -t $out/share/fonts/opentype/ otf/*.otf
+    runHook postInstall
+  '';
+
+
+  meta = with lib; {
+    description = "Intel One Mono Typeface";
+    homepage = "https://github.com/intel/intel-one-mono";
+    changelog = "https://github.com/intel/intel-one-mono";
```
Comment (felschr) [inline pkgs/data/fonts/intel-one-mono/default.nix:43]:
```suggestion
    changelog = "https://github.com/intel/intel-one-mono/releases/tag/V${version}";
```

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+
+  nativeBuildInputs = [
+    unzip
+  ];
+
+  unpackCmd = ''
+    mkdir srcs
+    cd srcs
```
Comment (felschr) [inline pkgs/data/fonts/intel-one-mono/default.nix:26]:
```suggestion
```

**Review by felschr — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{ lib, stdenvNoCC, fetchurl, unzip }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "intel-one-mono";
+  version = "1.0.0";
+
+  # currently v1.0.0 tag actually doesn't have any files, so using latest commit
```
Comment (felschr) [inline pkgs/data/fonts/intel-one-mono/default.nix:7]:
```suggestion
  # v1.0.0 tag actually doesn't have any files, so using release archives
```

**Review by felschr — APPROVED** [review]

**Review by drupol — APPROVED** [review]
LGTM!

**Comment by simoneruffini** [comment]
Hi, I'm the other guy who pushed pr #232467 , do you want to mantain it and move to version 1.2.0?

**Review by felschr — CHANGES_REQUESTED** [review]
Needs update to 1.2.0, which can also use `fetchFromGitHub` since the tagged release now contains `otf` & `ttf` folders.
Also please coordinate with author of #232467.

File: pkgs/data/fonts/intel-one-mono/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      url = "https://github.com/intel/intel-one-mono/releases/download/V${version}/otf.zip";
+      sha256 = "sha256-RxvLuWZiX4ojjw1jjLyWIqwW1RMGShJBlnmPYrxMgaw=";
+    })
+    (fetchurl {
+      url = "https://github.com/intel/intel-one-mono/releases/download/V${version}/ttf.zip";
+      sha256 = "sha256-skcETQOniKCHhfaj9RKOHxvDKmC93Aj9cudLG7K83bE=";
+    })
+  ];
+
+
```
Comment (felschr) [inline pkgs/data/fonts/intel-one-mono/default.nix:19]:
```suggestion

```

**Comment by dpc** [comment]
There's 1.2.0 already? Huh. 

I'm fine either way w.r.t. maintainership. Will try to update this one when I find some time if the other contributor isn't faster. If they are, please just land it.

**Comment by simoneruffini** [comment]
I'm doing it right now, I have some free time, if you want I can add you too to the mantainers.

**Comment by dpc** [comment]
No need. I'm more interested in using it, and just wanted it included.

**Comment by dpc** [comment]
Go with #232467 instead.

**Pull request closed without merging**